lib/src/context/base.dart

```diff
@@ -64,68 +66,191 @@ abstract class ICommandContextData implements IContextData {
+  /// - [parent], the context of which this context is the delegate.
+  IInteractiveContext? get delegate;
+
+  /// The youngest context that handles all interactions.
   ///
-  /// If [timeout] is set, this method will complete with an error after [timeout].
-  Future<IMultiselectInteractionEvent> getSelection(MultiselectBuilder selectionMenu,
-      {bool authorOnly = true, Duration? timeout = const Duration(minutes: 12)});
+  /// This is the same as repeatedly accessing [delegate] until it returns `null`.
+  IInteractiveContext get latestContext;
 
-  /// Wait for a user to press on a button, then return the result of that interaction.
+  /// Send a response to the command.
   ///
-  /// This method specifically listens for interactions on items of [buttons], ignoring other button
-  /// presses.
+  /// [level] can be set to change how the response is set. If is is not passed,
+  /// [CommandOptions.defaultResponseLevel] is used instead.
   ///
-  /// If [authorOnly] is `true`, only events triggered by the author of this context will be
-  /// returned, but other interactions will still be acknowledged.
+  /// You might also be interested in:
+  /// - [IInteractionInteractiveContext.acknowledge], for acknowledging interactions without
+  /// responding.
+  Future<IMessage> respond(MessageBuilder builder, {ResponseLevel? level});
+
+  /// Wait for a user to press a button and return a context representing that button press.
   ///
-  /// If [timeout] is set, this method will complete with an error after [timeout].
+  /// If [timeout] is set, this method will complete with an error after [timeout] has passed.
+  ///
+  /// If [authorOnly] is set, only the author of this interaction will be able to interact with the
+  /// button.
   ///
   /// You might also be interested in:
-  /// - [getConfirmation], a shortcut for getting user confirmation from buttons.
-  Future<IButtonInteractionEvent> getButtonPress(Iterable<ButtonBuilder> buttons,
-      {bool authorOnly = true, Duration? timeout = const Duration(minutes: 12)});
+  /// - [getSelection] and [getMultiSelection], for getting a selection from a user.
+  Future<ButtonComponentContext> getButtonPress(
+    String componentId, {
+    Duration? timeout = const Duration(minutes: 10),
+    bool authorOnly = true,
+  });
+
+  /// Wait for a user to select a single option from a multi-select menu and return a context
+  /// representing that selection.
+  ///
+  /// If [timeout] is set, this method will complete with an error after [timeout] has passed.
+  ///
+  /// If [authorOnly] is set, only the author of this interaction will be able to interact with the
+  /// selection menu.
+  ///
+  /// Will throw a [StateError] if more than one option is selected (for example, from a
+  /// multi-select menu allowing more than one choice).
+  Future<MultiselectComponentContext<T>> getSelection<T>(
+    String componentId, {
+    Duration? timeout = const Duration(minutes: 10),
```

Comment:
is nullable needed if default value provided?

Reply:
Previously, yes, it would have been (`null` = no timeout, so you would get errors if the user waited too long) but now it is not. Will remove.